templates/consumer/go/app/app.go

```diff
@@ -0,0 +1,60 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/irisnet/service-gen/service"
+	"github.com/irisnet/service-gen/{{service_name}}"
+	"github.com/irisnet/service-gen/types"
+	servicesdk "github.com/irisnet/service-sdk-go/service"
+	log "github.com/sirupsen/logrus"
+)
+
+// App represents the provider application
+type App struct {
+	ServiceClient    service.ServiceClientWrapper
+	ResponseCallback types.ResponseCallback
+	Logger           *log.Logger
+}
+
+// NewApp constructs a new App instance
+func NewApp(
+	serviceClient service.ServiceClientWrapper,
+	ResponseCallback types.ResponseCallback,
+	logger *log.Logger,
+) App {
+	return App{
+		ServiceClient:    serviceClient,
+		ResponseCallback: {{service_name}}.ResponseCallback,
+		Logger:           logger,
+	}
+}
+
+// Start starts the provider process
+func (app App) Start(addr string, responseCallback types.ResponseCallback) {
```

Comment:
no need the start function, move subscribe to Invoke.

Reply:
Let's keep it that way for the time being.